script.js

```diff
@@ -1,4 +1,122 @@
+
+// add a win loss record, output draws and winning percentage
+
+var STONE = `stone`;
+var SCISSORS = `scissors`;
+var PAPER = `paper`;
+var REVERSED_SCISSORS = `reversed scissors`;
+var REVERSED_PAPER = `reversed paper`;
+var REVERSED_STONE = `reversed stone`;
+
+var totalGamesWon = 0;
+var totalComWins = 0;
+var totalGamesPlayed = 0;
+var totalDrawGamesPlayed = 0;
+var userName = ``;
+var spsGame = `regular`
+var currentGameMode = `enter user name`
+
+var getWinRateInfo = function () {
+  var winRate = Math.floor(totalGamesWon/totalGamesPlayed *100);
+
+  if (totalGamesPlayed == 0) {
+    return `This is your first game!`
+  }
+  return `You Won ` +totalGamesWon + ` games.`
+  + `<br><br>Computer Won `+ totalComWins + ` games.`
+  +`<br></br>You drawed ${totalDrawGamesPlayed} games.`
+  +`<br><br>You win ` + winRate + `% of all games played.` 
+}
+
+// Assign a choice to computer via randomNumber function
+
+var generateChoice = function() {
+  var randomComChoice = randomNumber();
+  if (randomComChoice == 1) {
+    return SCISSORS;
+  }
+
+  if (randomComChoice == 2) {
+    return PAPER;
+  }
+
+  else { return STONE;
+  }
+
+}
+
+var randomNumber = function () {
+
+  var randomDecimal = Math.random() * 3;
+
+  var randomInterger = Math.floor(randomDecimal);
+
+  var randomNumber = randomInterger + 1;
+
+  return randomNumber;
+
+}
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
-};
+
+  var myOutputValue = `You lose, Try Again!`;
+
+  if (currentGameMode == `enter user name`) {
+
+    userName = prompt(`Please enter a username`);
```

Comment:
Does this line work like you intend it to? I notice you've invoked the syntax for executing a function (e.g. `prompt()`, but I don't see any helper function called prompt?